### DISCUSSION TITLE
FromComponentOn alternative in VContainer?

### DISCUSSION THREAD

**Discussion by kenofori** [hadashiA/VContainer#710] [Q&A]
Hi there,

I'm learning VContainer, and switching from Zenject,
I couldnt find alternative for "fromComponentOn", 

Container.Bind<MyClass>().FromComponentOn(sharedCanvas.gameObject).AsSingle();

please advise

**Comment by stenyin** [comment]
var sharedCanvasComponent = sharedCanvas.GetComponent<MyComponent>()
builder.RegisterInstance(sharedCanvasComponent);
